test/test_reductions.py

```diff
@@ -3145,6 +3146,7 @@ def reference_histogram(self, t, bins, bin_range, weights, density, dtype):
         self.assertEqual(hist_out, expected_hist)
         self.assertEqual(bin_edges_out, expected_bin_edges)
 
+    @unittest.expectedFailure  # histogram returns different values from Numpy in some cases, on the order of 2e-5 relative
```

Comment:
I'm assuming it only occurs with the non-contiguous inputs. It would not be ideal to xfail the entire test. If the relative error difference is consistent then it might be better to lift the tolerance in the [`assertEqual`](https://github.com/pytorch/pytorch/blob/c0e98a485ba32e95259860aabc379898aa141ff2/torch/testing/_internal/common_utils.py#L3805)?

Reply:
:+1: will do. The logic here was that the underlying problem appears to be that PyTorch and Numpy `histogram` et. al. produce substantively different values in a number of cases, so the XFAIL hopefully would call attention to the precision issues.

Reply:
@amjames Using a relative threshold now. I additionally reformatted some of the lines I was touching anyway to use a `black`-like style, so the diff looks a little large, but the comparison function is the only thing that got actually changed.